### ISSUE TITLE
twisted.python.log.DefaultObserver does its own weird thing, doesn't use textFromEventDict

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6569]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#6569|
|Type|defect|
|Created|2013-06-11 18:26:36Z|
|Branch|https://github.com/twisted/twisted/tree/log-default-observer-6569|

Among other things, this means that `DefaultObserver` does not safely `repr` its argument.  So, logging stuff like `log.msg("Hello", BrokenRepr(), isError=1)` will result in an exception and `DefaultLogObserver` being removed, whereas `FileLogObserver` will handle it correctly.

<details><summary>Searchable metadata</summary>

```
trac-id__6569 6569
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_log_default_observer_6569 branches/log-default-observer-6569
branch_author__wsanchez wsanchez
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1370975196000000 1370975196000000
changetime__1429135553159331 1429135553159331
version__None None
owner__wsanchez wsanchez

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez set status to `assigned`|
|-|-|

Accepting, holding for [#6750](https://github.com/twisted/twisted/issues/6750), which should either fix this or make it cleaner to fix.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_2"></a>|@wsanchez commented|
|-|-|

(In [44467]) Branch for [#6569](https://github.com/twisted/twisted/issues/6569).

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez removed owner<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm set owner to @ralphm<br>@ralphm set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_5"></a>|@ralphm set owner to @wsanchez<br>@ralphm set status to `new`|
|-|-|

Yay for removing duplicate code. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez set status to `closed`|
|-|-|

(In [44476]) Merge branches/log-default-observer-6569: Don't raise if DefaultObserver.emit() gets a hostile message.

Author: wsanchez
Reviewer: ralphm
Fixes: [#6569](https://github.com/twisted/twisted/issues/6569)
